mev_inspect/sandwiches.py

```diff
@@ -42,32 +42,57 @@ def _get_sandwich_starting_with_swap(
     ]:
         return None
 
-    for other_swap in rest_swaps:
-        if other_swap.transaction_hash == front_swap.transaction_hash:
+    for back_swap in rest_swaps:
+        if back_swap.transaction_hash == front_swap.transaction_hash:
             continue
 
-        if other_swap.contract_address == front_swap.contract_address:
+        if back_swap.contract_address == front_swap.contract_address:
             if (
-                other_swap.token_in_address == front_swap.token_in_address
-                and other_swap.token_out_address == front_swap.token_out_address
-                and other_swap.from_address != sandwicher_address
+                back_swap.token_in_address == front_swap.token_in_address
+                and back_swap.token_out_address == front_swap.token_out_address
+                and back_swap.from_address != sandwicher_address
             ):
-                sandwiched_swaps.append(other_swap)
+                sandwiched_swaps.append(back_swap)
             elif (
-                other_swap.token_out_address == front_swap.token_in_address
-                and other_swap.token_in_address == front_swap.token_out_address
-                and other_swap.from_address == sandwicher_address
+                back_swap.token_out_address == front_swap.token_in_address
+                and back_swap.token_in_address == front_swap.token_out_address
+                and back_swap.from_address == sandwicher_address
             ):
                 if len(sandwiched_swaps) > 0:
+                    profit_amount: float
+                    if back_swap.token_in_amount == 0 and back_swap.error is None:
+                        raise ValueError("Backrun cannot swap 0 tokens")
```

Comment:
This error doesn't have a test case.

Reply:
we don't expect this to ever happen on chain; i added this here to be explicit about the assumption that the back swap must be ruled out in the case this value is 0 as to avoid a DivisionByZero error, so test coverage here would not be meaningful